src/sctools/metrics/metricsrunner.py

```diff
@@ -0,0 +1,61 @@
+import argparse
+import pysam
+
+
+class MetricsRunner:
+
+    def __init__(self, sam_file, open_mode=None):
+        """
+        :param str alignment_file: sam or bam file.
+        :param str open_mode: optional, mode to read file. Will be autodetected by file type if
+          the file contains the correct suffix for its type.
+        """
+
+        if open_mode is None:
+            if sam_file.endswith('.bam'):
+                open_mode = 'rb'
+            elif sam_file.endswith('.sam'):
+                open_mode = 'r'
+            else:
+                raise ValueError('could not autodetect file type for alignment_file %s '
+                                 '(detectible suffixes: .sam, .bam)' % sam_file)
+        self.input_sam = sam_file
+        self.open_mode = open_mode
+
+    def run_metrics(self, metrics_to_run):
+        """
+        :param list metrics_to_run: list of metrics to run for each record in input_sam
+        """
+
+        for metric in metrics_to_run:
+            metric.setup()
+
+        with pysam.AlignmentFile(self.input_sam, self.open_mode) as sam:
+            for record in sam:
+                for metric in metrics_to_run:
+                    metric.gather_metric(record)
+
+        for metric in metrics_to_run:
+            metric.calculate_and_output()
+
+
+def convert_class_name_to_class(metric_class_names):
```

Comment:
Good idea. I think this is just `getattr(module, name)` or `operator.attrgetter` if we want it to have a custom name. 

https://docs.python.org/3.7/library/operator.html#operator.attrgetter